File: server/router/system/email_verify.go
package system

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type EmailVerifyRouter struct{}

func (r *EmailVerifyRouter) InitEmailVerifyRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	group := Router.Group("emailVerify").Use(middleware.OperationRecord())
	groupWithoutRecord := Router.Group("emailVerify")
	{
		group.POST("start", emailVerifyApi.StartEmailVerify)
	}
	{
		groupWithoutRecord.GET("job", emailVerifyApi.GetEmailVerifyJob)
		groupWithoutRecord.GET("history", emailVerifyApi.GetEmailVerifyHistory)
	}
}

File: web/src/api/emailVerify.js
import service from '@/utils/request'

export const startEmailVerify = (data) => {
  return service({
    url: '/emailVerify/start',
    method: 'post',
    data
  })
}

export const getEmailVerifyJob = (params) => {
  return service({
    url: '/emailVerify/job',
    method: 'get',
    params
  })
}

export const getEmailVerifyHistory = (params) => {
  return service({
    url: '/emailVerify/history',
    method: 'get',
    params
  })
}

